### PR TITLE
Selenium: Add new test method to the 'PullRequestPluginTest'

### PR DESCRIPTION
**Pull request by artaleks9**
### What does this PR do?
* Update the selenium test _PullRequestPluginTest_
  - add new test method _createPullRequestToNonDefaultBranch()_ to the test class
  - rework the _switchingBetweenProjects()_ test method
  - rework and rename the _createPullRequest()_ test method
  - refactor code
*  Add the method 'clickRefreshContributionBranchButton()' to the _PullRequestPanel_
* Add the id selector to the web element in the _ContributePartViewImpl_
* Delete the 'try/catch' from the _PullRequestPluginWithForkTest_ related to close the issue

### What issues does this PR fix or reference?
#10000

### CODE REVIEW

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/759/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/759//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/11128)
docker image: eclipseche/che-server:11128
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by RomanNikitenko — APPROVED** [review]

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/PullRequestPanel.java [hunk 1]
```diff
@@ -64,6 +64,7 @@ public PullRequestPanel(
     static final String PULL_REQUEST_BTN = "gwt-debug-partButton-Pull Request";
     static final String BRANCH_NAME =
         "//div[text()='Branch name:']/following-sibling::div//span/label[text()='%s']";
+    static final String REFRESH_BRANCH_BUTTON = "gwt-debug-refreshContributionBranchButton";
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/PullRequestPanel.java:67]: Is it button name or id? Could you express it in the constant name to make it clear?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/PullRequestPanel.java [hunk 2]
```diff
@@ -195,6 +199,11 @@ public void selectBranch(String branchName) {
         By.xpath(String.format(PullRequestLocators.BRANCH_NAME, branchName)));
   }
 
+  /** Click 'refresh contribution branch name' button on the 'Pull Request' panel */
+  public void clickRefreshContributionBranchButton() {
+    seleniumWebDriverHelper.waitAndClick(refreshBranchListButton);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/PullRequestPanel.java:204]: `refreshBranchListButton` variable name hasn't completely agreed with button name expressed in method name `refreshContributionBranchButton`

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java [hunk 3]
```diff
@@ -64,10 +71,17 @@
   private static final String MAIN_BRANCH = "master";
   private static final String PULL_REQUEST_CREATED = "Your pull request has been created.";
   private static final String PUll_REQUEST_UPDATED = "Your pull request has been updated.";
-  private static final String NEW_BRANCH = generate("branch-", 8);
+  private static final String DEV_BRANCH_NAME = generate("dev-", 4);
+  private static final String BASE_BRANCH_NAME = generate("contrib-", 4);
+  private static final String NEW_BRANCH_NAME = generate("branch-", 8);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java]: `NEW_BRANCH_NAME = generate("new-", 4);` looks more suitable here

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java [hunk 4]
```diff
@@ -231,12 +296,29 @@ private void configureTypeOfProject() {
...
   }
 
+  /** check the target branches in the pull header */
+  private void checkBranchNamesOnGitHubPage() {
+    String pullHeaderText =
+        format(
+            "%s wants to merge 1 commit into %s\n" + "from %s",
+            gitHubUsername, BASE_BRANCH_NAME, DEV_BRANCH_NAME);
+    seleniumWebDriverHelper.waitTextContains(
+        By.xpath("//div[@class='TableObject-item TableObject-item--primary']"), pullHeaderText);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java]: It's better to name `//div[@class='TableObject-item TableObject-item--primary']` somehow to understand it's propose.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java [hunk 5]
```diff
@@ -114,65 +135,106 @@ public void setUp() throws Exception {
...
 
-    // change commit and create pull request
+    // commit the change create pull request
+    pullRequestPanel.clickCreatePullRequestButton();
+    pullRequestPanel.clickOkCommitBtn();
+    pullRequestPanel.waitStatusOk(BRANCH_PUSHED_ON_YOUR_ORIGIN);
+    pullRequestPanel.waitStatusOk(PULL_REQUEST_ISSUED);
+    pullRequestPanel.waitMessage(PULL_REQUEST_CREATED);
+
+    // check the pull header on the GitHub page
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PullRequestPluginTest.java]: it's not just pull header, but Pull Request header

**Comment by dmytro-ndp** [comment]
IMHO checking pull request through the GitHub API by using is more reliably and safe method compare to getting to GitHub pull request web page and checking web elements state by it's xpath.
We could use `org.kohsuke.github.GHRepository.getPullRequest()` method for this propose.

**Comment by artaleks9** [comment]
OK, it's a good idea

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/840/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/840//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11128
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/841/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/841//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11128
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by musienko-maxim — APPROVED** [review]

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/870/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/870//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11128
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by dmytro-ndp** [comment]
@artaleks9: please, review new changes.

**Comment by dmytro-ndp** [comment]
ci-test

**Comment by artaleks9** [comment]
Looks good for me

**Comment by riuvshin** [comment]
Results of automated E2E tests of Eclipse Che Multiuser on OCP:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/896/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/896//Selenium_20tests_20report/)
docker image: eclipseche/che-server:11128
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**